dask_expr/collection.py

```diff
@@ -85,6 +85,13 @@ def _meta(self):
     def size(self):
         return new_collection(self.expr.size)
 
+    def __len__(self):
+        return self._len
+
+    @functools.cached_property
+    def _len(self):
+        return sum(new_collection(expr.Lengths(self.expr)).compute())
```

Comment:
I think that we'll still need to use `Len` here.  

I'm not sure how I feel about caching.  Ok to leave this as-is for now, and this probably makes sense to do, but it does open some questions about caching more generally.

Reply:
Went back to using `Len` and removed the caching.